### PR TITLE
Neat 10 implement rules information architect

### PR DESCRIPTION
**Pull request by nikokaoja**

### CODE REVIEW

File: cognite/neat/rules/models/_rules/domain_rules.py [hunk 1]
```diff
@@ -1,28 +1,130 @@
...
 
 class DomainMetadata(BaseMetadata):
     role: ClassVar[RoleTypes] = RoleTypes.domain_expert
     creator: str | list[str]
 
+    @field_validator("creator", mode="before")
+    def creator_to_list_if_comma(cls, value):
+        if isinstance(value, str):
+            if value:
+                return value.replace(", ", ",").split(",")
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/domain_rules.py:36]:
```suggestion
        if isinstance(value, str) and value:
            return value.replace(", ", ",").split(",")
```

File: cognite/neat/rules/models/_rules/domain_rules.py [hunk 2]
```diff
@@ -1,28 +1,130 @@
...
+    class_: ExternalId = Field(alias="Class", min_length=1, max_length=255)
+    property: ExternalId = Field(alias="Property", min_length=1, max_length=255)
+    value_type: ValueType = Field(alias="Value Type")
+    min_count: int | None = Field(alias="Min Count", default=None)
+    max_count: int | float | None = Field(alias="Max Count", default=None)
+
+    @field_validator("value_type", mode="before")
+    def expected_value_type_string_to_entity(cls, value):
+        # handle simple types
+        if value in XSD_VALUE_TYPE_MAPPINGS.keys():
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/domain_rules.py:50]:
```suggestion
        if value in XSD_VALUE_TYPE_MAPPINGS:
```

File: cognite/neat/rules/models/_rules/domain_rules.py [hunk 3]
```diff
@@ -1,28 +1,130 @@
...
+        if not re.match(class_id_compliance_regex, value):
+            raise exceptions.PropertiesSheetClassIDRegexViolation(
+                value, class_id_compliance_regex
+            ).to_pydantic_custom_error()
+        else:
+            return value
 
-class DomainClass(Entity):
+    @field_validator("property", mode="after")
+    def is_property_id_compliant(cls, value, values):
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/domain_rules.py:73]: Not used.
```suggestion
    def is_property_id_compliant(cls, value):
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 4]
```diff
@@ -1,20 +1,108 @@
+import re
+import warnings
 from datetime import datetime
-from typing import ClassVar
+from enum import StrEnum
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:4]:
```suggestion
import sys
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 5]
```diff
@@ -1,20 +1,108 @@
...
 
+from .base import (
+    Prefix,
+    RoleTypes,
+    RuleModel,
+    SheetList,
+    prefix_compliance_regex,
+    version_compliance_regex,
+)
+from .domain_rules import DomainClass, DomainMetadata, DomainProperty
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:29]:
```suggestion
from .domain_rules import DomainClass, DomainMetadata, DomainProperty

if sys.version_info >= (3, 11):
    from enum import StrEnum
else:
    from backports.strenum import StrEnum
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 6]
```diff
@@ -1,20 +1,108 @@
...
 
-class InformationArchitectMetadata(DomainMetadata):
+
+class MatchType(StrEnum):
+    exact = "exact"
+    partial = "partial"
+
+
+class InformationMetadata(DomainMetadata):
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:37]: Suggest we inherit from the base metadata and not domain.

We do not want the creator on metadata.
```suggestion
class InformationMetadata(BaseMetadata):
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 7]
```diff
@@ -1,20 +1,108 @@
...
+    updated: datetime = Field(
+        description=("Date of the data model update"),
+    )
+
+    @field_validator("version")
+    def is_version_compliant(cls, value):
+        if not re.match(version_compliance_regex, value):
+            raise exceptions.VersionRegexViolation(value, version_compliance_regex).to_pydantic_custom_error()
+        else:
+            return value
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:82]: Just my personal preference but I find it more readable

If this -> raise exception.


```suggestion
        if not re.match(version_compliance_regex, value):
            raise exceptions.VersionRegexViolation(value, version_compliance_regex).to_pydantic_custom_error()
        return value
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 8]
```diff
@@ -1,20 +1,108 @@
...
+            raise exceptions.PrefixesRegexViolation([value], prefix_compliance_regex).to_pydantic_custom_error()
+        else:
+            return value
+
+    @field_validator("contributor", mode="before")
+    def contributor_to_list_if_comma(cls, value):
+        if isinstance(value, str):
+            if value:
+                return value.replace(", ", ",").split(",")
+        return value
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:96]:
```suggestion
        if isinstance(value, str) and value:
            return value.replace(", ", ",").split(",")
        return value
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 9]
```diff
@@ -1,20 +1,108 @@
...
+            raise exceptions.VersionRegexViolation(value, version_compliance_regex).to_pydantic_custom_error()
+        else:
+            return value
+
+    @field_validator("prefix")
+    def is_prefix_compliant(cls, value):
+        if not re.match(prefix_compliance_regex, value):
+            raise exceptions.PrefixesRegexViolation([value], prefix_compliance_regex).to_pydantic_custom_error()
+        else:
+            return value
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:89]:
```suggestion
        if not re.match(prefix_compliance_regex, value):
            raise exceptions.PrefixesRegexViolation([value], prefix_compliance_regex).to_pydantic_custom_error()
        return value
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 10]
```diff
@@ -26,6 +114,7 @@ def from_domain_expert_metadata(
         contributor: str | list[str] | None = None,
         created: datetime | None = None,
         updated: datetime | None = None,
+        name: str | None = None,
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:117]: This is unused
```suggestion
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 11]
```diff
@@ -37,5 +126,65 @@ def from_domain_expert_metadata(
...
 
-class InformationArchitectRules(RuleModel):
-    metadata: InformationArchitectMetadata
+class InformationClass(DomainClass):
+    """
+    Class is a category of things that share a common set of attributes and relationships.
+
+    Args:
+        class_: The class ID of the class.
+        name: The name of the class.
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:135]:
```suggestion
        description: A description of the class.
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 12]
```diff
@@ -37,5 +126,65 @@ def from_domain_expert_metadata(
...
+
+class InformationProperty(DomainProperty):
+    """
+    A property is a characteristic of a class. It is a named attribute of a class that describes a range of values
+    or a relationship to another class.
+
+    Args:
+        class_: Class ID to which property belongs
+        property: Property ID of the property
+        name: Property name.
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:159]: I haven't seen this is the sheets we have.
```suggestion
```

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 13]
```diff
@@ -37,5 +126,65 @@ def from_domain_expert_metadata(
...
+    def fix_namespace_ending(cls, value):
+        if value:
+            return Namespace(TypeAdapter(HttpUrl).validate_python(value))
+        return value
+
+
+class InformationRules(RuleModel):
+    metadata: InformationMetadata = Field(alias="Metadata")
+    properties: SheetList[DomainProperty] = Field(alias="Properties")
+    classes: SheetList[DomainClass] = Field(alias="Classes")
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/information_rules.py:190]:
```suggestion
    properties: SheetList[InformationProperty] = Field(alias="Properties")
    classes: SheetList[InformationClass] = Field(alias="Classes")
```

**Review by doctrino — COMMENTED** [review]
Some minor suggestions. And then, setting correct classes for InformaionRule.

File: cognite/neat/rules/models/_rules/information_rules.py (hunk 12 above)
Comment (nikokaoja) [reply cognite/neat/rules/models/_rules/information_rules.py:162]: There is, name is used by a customer who has cryptic classes (class -> class_id) where they need human readable name , thus we have name added

**Review by doctrino — APPROVED** [review]

**Pull request merged**